staff/clara-rufi/Precurs2019/PasapalabraTimer/Pasapalabra.js

```diff
@@ -0,0 +1,413 @@
+    { letter: "y", answer: "peyote", status: 0, question: "CONTIENE LA Y. Pequeño cáctus conocido por sus alcaloides psicoactivos utilizado de forma ritual y medicinal por indígenas americanos"},
+    { letter: "z", answer: "zen", status: 0, question: "CON LA Z. Escuela de budismo que busca la experiencia de la sabiduría más allá del discurso racional"},
+]
+ 
+
+
+
+
+
+// Pasapalabra
+
+nameUser = []
+    document.getElementById('start').style="display:none;"
+    document.getElementById('letters').style="display:none;"
+    document.getElementById('showQuestion').style="display:none;"
+    document.getElementById('userAnswer').style="display:none;"
+    document.getElementById('acceptar').style="display:none;"
+    document.getElementsByClassName('letter').style="display:none;"
+    document.getElementById('Okrepeat').style="display:none;"//pq no es mostri
+    document.getElementById('repeatPasapalabra').style="display:none;"
+    document.getElementById('showresults').style="display:none;"
+    document.getElementById('results').style="display:none;"
+    document.getElementById("end").style="display:none;" 
+    document.getElementById("questionPlayAgain").style="display:none;"
+    document.getElementById("playAgainYes").style="display:none;"
+    document.getElementById("playAgainNo").style="display:none;"
+    document.getElementById("goodBye").style="display:none;"
+    document.getElementById("time").style="display:none;"
+    document.getElementById("tiempo").style="display:none;"
+    document.getElementById("minuts").style="display:none;"
+    document.getElementById("timeFinished").style="display:none;"
+
+function login(){
+
+var name = document.getElementById("nameUser").value
+document.getElementById("welcome").value = ""
+    if (name.length >=2){
+        nameUser.push(name)
+        console.log(nameUser)
+        document.getElementById("welcome").value += "Bienvenido " + nameUser + " !!"
+        document.getElementById('start').style="display:block;"
+    }else{
+        document.getElementById("welcome").value += "Nombre válido mínimo 2 carácteres"
+        document.getElementById("nameUser").value = ""
+    }
+    
+}
+
+
+n = 0
+function start(){
+    document.getElementById('letters').style="display:block;"
+    document.getElementById('showQuestion').style="display:block;"
+    document.getElementById('userAnswer').style="display:block;"
+    document.getElementById('loginUser').style="display:none;" 
+    document.getElementById('acceptar').style="display:block;"
+    document.getElementById('end').style="display:block;"
+    document.getElementById("time").style="display:block;"
+    document.getElementById("tiempo").style="display:block;"
+    document.getElementById("minuts").style="display:block;"
+    timer();
+    document.getElementById("showQuestion").value = questions[n].question;
+}
+
+// timer 
+
+function timer(){
+    startTimer()
+}
+
+function startTimer(duration, display) {
```

Comment:
In order to the countdown works you should do some changes in `startTimer()` function. I show you an example:

```javascript
function startTimer(duration, display) {
    
    var __timer = setInterval(function () {
                
        minutes = Math.floor(duration / 60);
        seconds = duration-Math.round(minutes *60);

        minutes = minutes < 10 ? "0" + minutes : minutes;
        seconds = seconds < 10 ? "0" + seconds : seconds;

        display.textContent = minutes + ":" + seconds;

        if (duration <= 0) {
            clearInterval(__timer)
            document.getElementById("timeFinished").style="display:block;"
            document.getElementById("timeFinished").value = "Tiempo acabado"
            document.getElementById("time").style="display:none;"
            document.getElementById("tiempo").style="display:none;"
            document.getElementById("minuts").style="display:none;"
            document.getElementById('letters').style="display:none;"
            document.getElementById('showQuestion').style="display:none;"
            document.getElementById('userAnswer').style="display:none;"
            document.getElementById('acceptar').style="display:none;"
            document.getElementsByClassName('letter').style="display:none;"
            document.getElementById('Okrepeat').style="display:none;"//pq no es mostri
            document.getElementById('repeatPasapalabra').style="display:none;"
            document.getElementById('showresults').style="display:none;"
            document.getElementById('results').style="display:none;"
            document.getElementById("end").style="display:none;" 
            document.getElementById("questionPlayAgain").style="display:block;"
            document.getElementById("playAgainYes").style="display:block;"
            document.getElementById("playAgainNo").style="display:block;"
           
        }
        duration--;
    }, 1000);
}
```

If you want tomorrow we can speak about this. Before doing the change you should also change all `startTimer` calls. I will try to explain you this in the following comments.

Reply:
ok thanks. in fact, i found another countdown and i wanted to try it. but thanks for your suggestions, i will consider them, too :)